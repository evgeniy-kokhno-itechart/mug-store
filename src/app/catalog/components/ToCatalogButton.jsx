import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

const ToCatalogButton = ({ customClasses }) => (
  <Link className={`button button-solid ${customClasses}`} to="/catalog">
    Back To Catalog
  </Link>
);

ToCatalogButton.propTypes = {
  customClasses: PropTypes.string,
};

ToCatalogButton.defaultProps = {
  customClasses: '',
};

export default ToCatalogButton;
