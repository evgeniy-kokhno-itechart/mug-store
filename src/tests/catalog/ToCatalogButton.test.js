/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ToCatalogButton from '../../catalog/components/ToCatalogButton';

configure({ adapter: new Adapter() });

describe('<ToCatalogButton />', () => {
  let ToCatalogButtonWrapper;

  beforeEach(() => {
    ToCatalogButtonWrapper = shallow(<ToCatalogButton />);
  });

  it('check render without customClasses provided', () => {
    expect(ToCatalogButtonWrapper.debug()).toMatchSnapshot();
  });

  it('check render with customClasses = testclass provided', () => {
    ToCatalogButtonWrapper.setProps({ customClasses: 'testclass' });
    expect(ToCatalogButtonWrapper.debug()).toMatchSnapshot();
  });
});
