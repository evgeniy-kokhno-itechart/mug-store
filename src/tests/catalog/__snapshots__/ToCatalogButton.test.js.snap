// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToCatalogButton /> check render with customClasses = testclass provided 1`] = `
<Link
  className="button button--solid testclass"
  to="/catalog"
>
  Back To Catalog
</Link>
`;

exports[`<ToCatalogButton /> check render without customClasses provided 1`] = `
<Link
  className="button button--solid "
  to="/catalog"
>
  Back To Catalog
</Link>
`;
